module/VuFind/src/VuFind/Controller/AbstractRecord.php

```diff
@@ -492,13 +493,14 @@ public function saveAction()
 
         // Find out if the item is already part of any lists; save list info/IDs
         $listIds = [];
-        $resources = $user->getSavedData(
+        $resources = $this->getDbService(UserResourceServiceInterface::class)->getFavoritesForRecord(
             $driver->getUniqueId(),
+            $driver->getSourceIdentifier(),
             null,
-            $driver->getSourceIdentifier()
+            $user
         );
         foreach ($resources as $userResource) {
-            $listIds[] = $userResource->list_id;
+            $listIds[] = $userResource->getUserList()->getId();
```

Comment:
`null` is a legal return value for `getUserList()`, do we need to take that into account here?